metis/lib/server/controllers/metis_controller.rb

```diff
@@ -36,14 +36,15 @@ def require_folder(bucket, folder_path)
 
     def file_hashes_with_calculated_paths(bucket:, files:)
       # Calculate the folder_path, instead of
-      #   doing it in the database.
+      #   doing it in the database because there is no bulk way to do that,
+      #   and is recursive per file. Try to save queries / connections
       folder_path_calc = Metis::FolderPathCalculator.new(bucket: bucket)
 
       return [] unless files
 
       files.map { |file|
-        if file.folder
-          path = "#{folder_path_calc.get_folder_path(file.folder)}/#{file.file_name}"
+        if file.folder_id
```

Comment:
My working assumption is that calling `file.folder` caused a database query, so each time a file was expanded to have it's path calculated, the database was hit. Using just the `folder_id` should circumvent that, and reduce contention for db connections...

Reply:
True.  I think this change will in fact help.